specs/authentication_and_authorisation_linkage_be.adoc

```diff
@@ -0,0 +1,124 @@
+# Authentication and Authorisation (AA) linkage between tendrl UI and Grafana
+
+The intent of this spec is to:-
+
+  - Sync read-write and read-only roles per user between tendrl and grafana, priority being read-only.
+  - Sync user creation and changes to existing users and their roles in tendrl to grafana.
+  - Users authenticated via tendrl should be automatically authenticated in grafana as well.
+  - Launching a cluster specific grafana dashboard should be pre-authorised from within tendrl ui.
+
+
+## Problem Description
+
+Since the Tendrl stack is going to include Grafana for monitoring, so there will be a button in UI of Tendrl which can take users to Grafana Dashboard to view different charts.Tendrl has an authentication system and Grafana has there own authentication system as well. So instead of having two authentication for users, we are planning to bring it down to one.
+  
+## Use Cases
+Provide seemless authorization between two dashboards.
+There are three types of users in Tendrl:- 
+* Admin
+* Normal
+* Limited
+
+There are four types of user in Grafana:-
+* Admin
+* Editor
+* Read-only Editor
+* Viewer
+
+So the roles can be mapped as follows:-
+
+Admin <----------------------> Admin
+
+Normal <----------------------> Editor
+
+Limited <----------------------> Read-only Editor / Viewer
+
+## Proposed change
+
+The Authentication seamlessness by enabling anonymous login
+```
+[auth.anonymous]
+enabled = true
+org_name = YourOrgName
+org_role = Viewer/Editor/Read-only editor
+```
+For every user there will be an organization created in grafana
```

Comment:
Based on recent discussions in daily meetings, there needs to be some changes I feel here. An org for every user is the point I am emphasizing on.

Reply:
Yes, I need to update this